pgcli/packages/parseutils.py

```diff
@@ -188,6 +188,43 @@ def find_prev_keyword(sql):
 
     return None, ''
 
+
+# Postgresql dollar quote signs look like `$$` or `$tag$`
+dollar_quote_regex = re.compile(r'^\$[^$]*\$$')
+
+
+def is_open_quote(sql):
+    """Returns true if the query contains an unclosed quote"""
+
+    # parsed can contain one or more semi-colon separated commands
+    parsed = sqlparse.parse(sql)
+    return any(_parsed_is_open_quote(p) for p in parsed)
+
+
+def _parsed_is_open_quote(parsed):
+    tokens = list(parsed.flatten())
+
+    i = 0
+    while i < len(tokens):
+        tok = tokens[i]
+        if tok.match(Token.Error, "'"):
+            # An unmatched single quote
+            return True
+        elif (tok.ttype in Token.Name.Builtin
+                and dollar_quote_regex.match(tok.value)):
+            # Find the matching closing dollar quote sign
+            for (i, tok2) in enumerate(tokens[i+1:], i+1):
```

Comment:
I'm not a fan of reusing the index variable `i` here. It utilizes one of the obscure scoping rules of Python and it's not readily apparent. 

Can you change this to use `j` and then increment it to `i` as `i += j` in that `elif` clause?